frontend/public/components/storage/access-modes/access-mode.tsx

```diff
@@ -26,62 +22,64 @@ export const getPVCAccessModes = (resource: PersistentVolumeClaimKind, key: stri
 
 export const AccessModeSelector: React.FC<AccessModeSelectorProps> = (props) => {
   const {
-    formClassName,
+    className,
     pvcResource,
+    filterByVolumeMode,
     onChange,
     loaded,
     provisioner,
     availableAccessModes = [],
+    description,
+    ignoreReadOnly,
   } = props;
 
   const { t } = useTranslation();
-
   const pvcInitialAccessMode = pvcResource
     ? getPVCAccessModes(pvcResource, 'value')
     : availableAccessModes;
 
   const [allowedAccessModes, setAllowedAccessModes] = React.useState<string[]>();
-  const [accessMode, setAccessMode] = React.useState<string>('');
+  const [accessMode, setAccessMode] = React.useState<string>();
 
-  const updateAllowedAccessModes = (scProvisioner: string) =>
-    provisionerAccessModeMapping[scProvisioner] ?? getAccessModeForProvisioner('');
+  const changeAccessMode = React.useCallback(
+    (mode: string) => {
+      setAccessMode(mode);
+      onChange(mode);
+    },
+    [onChange],
+  );
 
   React.useEffect(() => {
     if (loaded) {
-      let currentModes: string[];
-      provisioner?.includes(cephRBDProvisionerSuffix) && pvcResource?.spec.volumeMode !== 'Block'
-        ? (currentModes = ['ReadWriteOnce', 'ReadOnlyMany'])
-        : (currentModes = updateAllowedAccessModes(provisioner));
-      setAllowedAccessModes(currentModes);
+      setAllowedAccessModes(
+        getAccessModeForProvisioner(
+          provisioner,
+          ignoreReadOnly,
+          filterByVolumeMode && pvcResource?.spec?.volumeMode,
```

Comment:
Won't think sometime return a boolean and sometime a string? It would be better to do something like

```suggestion
          filterByVolumeMode ? pvcResource?.spec?.volumeMode : null,
```

Reply:
let me check this